codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectEncoder.java

```diff
@@ -331,7 +351,7 @@ private void encodeFullHttpMessage(ChannelHandlerContext ctx, Object o, List<Obj
     private static boolean encodeContentNonChunk(List<Object> out, ByteBuf buf, ByteBuf content) {
         final int contentLength = content.readableBytes();
         if (contentLength > 0) {
-            if (buf.writableBytes() >= contentLength) {
+            if (buf.maxWritableBytes() >= contentLength) {
```

Comment:
What is the value of `maxWritableBytes()` typically? I worry that it might be on the order of `Integer.MAX_VALUE`.

Reply:
It should be the max without any resizing, which include some padding due to allocations: I remember @njhill proposed it

Reply:
Did you intend to use `maxFastWritableBytes()`?

Reply:
Good catch! Yep! I will fix it, thanks!